assets/quick-order-list.js

```diff
@@ -1,307 +1,279 @@
-          const debouncedOnChange = debounce((event) => {
-            this.onChange(event);
-          }, 100);
-          qty.addEventListener('change', debouncedOnChange.bind(this));
-        });
+      toggleTableLoading(enable) {
+        this.querySelector('.quick-order-list__table').classList.toggle(
+          'quick-order-list__container--disabled',
+          enable
+        );
+        this.toggleLoading(enable);
       }
 
-      renderSections(parsedState, ids) {
-        this.ids.push(ids);
-        const intersection = this.queue.filter((element) => ids.includes(element.id));
-        if (intersection.length !== 0) return;
+      async refresh(pageNumber = null) {
+        const url = this.dataset.url || window.location.pathname;
 
-        this.getSectionsToRender().forEach((section) => {
-          const sectionElement = document.getElementById(section.id);
-          if (
-            sectionElement &&
-            sectionElement.parentElement &&
-            sectionElement.parentElement.classList.contains('drawer')
-          ) {
-            parsedState.items.length > 0
-              ? sectionElement.parentElement.classList.remove('is-empty')
-              : sectionElement.parentElement.classList.add('is-empty');
-            setTimeout(() => {
-              document.querySelector('#CartDrawer-Overlay').addEventListener('click', this.cart.close.bind(this.cart));
-            });
-          }
-          const elementToReplace =
-            sectionElement && sectionElement.querySelector(section.selector)
-              ? sectionElement.querySelector(section.selector)
-              : sectionElement;
-          if (elementToReplace) {
-            if (section.selector === `#${this.quickOrderListId} .js-contents` && this.ids.length > 0) {
-              this.ids.flat().forEach((i) => {
-                elementToReplace.querySelector(`#Variant-${i}`).innerHTML = this.getSectionInnerHTML(
-                  parsedState.sections[section.section],
-                  `#Variant-${i}`
-                );
-              });
-            } else {
-              elementToReplace.innerHTML = this.getSectionInnerHTML(
-                parsedState.sections[section.section],
-                section.selector
+        return fetch(`${url}?section_id=${this.dataset.section}&page=${pageNumber || this.currentPage}`)
+          .then((response) => response.text())
+          .then((responseText) => {
+            const html = new DOMParser().parseFromString(responseText, 'text/html');
+            const responseQuickOrderList = html.querySelector(`#${this.id}`);
+
+            if (!responseQuickOrderList) {
+              return;
+            }
+
+            this.innerHTML = responseQuickOrderList.innerHTML;
+            this.initEventListeners();
+          })
+          .catch((e) => {
+            console.error(e);
+          });
+      }
+
+      renderSections(parsedState) {
```

Comment:
Primary change here is to how the quickorderlist section is updated to account for pagination. Fixed a few bugs too:
- focus state is now preserved after update
- no longer ignores cart updates to this product made from other tabs